components/text-field/script.js

```diff
@@ -0,0 +1,504 @@
+     */
+    clearMousedownInterval() {
+        document.removeEventListener('mouseup', this.clearMousedownIntervalBound);
+
+        if (this.mousedownTimeout) {
+            clearTimeout(this.mousedownTimeout);
+            this.mousedownTimeout = null;
+        }
+
+        if (this.mousedownInterval) {
+            clearInterval(this.mousedownInterval);
+            this.mousedownInterval = null;
+        }
+    }
+
+    /**
+     * Adds an interval after a small timeout that will constantly bump up or down the number input value until the mouse is released
+     * from the arrow of the input control
+     * @param {Function} stepCallback - Should be some of stepUp or stepDown functions.
+     */
+    addAutoStepInterval(stepCallback) {
+        document.addEventListener('mouseup', this.clearMousedownIntervalBound);
+
+        if (!this.mousedownTimeout) {
+            this.mousedownTimeout = setTimeout(() => {
+                if (!this.mousedownInterval) {
+                    this.mousedownInterval = setInterval(stepCallback.bind(this), 50);
+                }
+            }, 200)
+        }
+    }
+
+    /**
+     * Will set the auto step interval and bump up the input value
+     * @param {Event} event
+     */
+    onStepUp(event) {
+        this.addAutoStepInterval(this.stepUp);
+        this.stepUp();
+        event.preventDefault();
+    }
+
+    /**
+     * Will set the auto step interval and bump down the input value
+     * @param {Event} event
+     */
+    onStepDown(event) {
+        this.addAutoStepInterval(this.stepDown);
+        this.stepDown();
+        event.preventDefault();
+    }
+
+    /**
+     * Will bump up the input value based on the min and max restrinctions
+     */
+    stepUp() {
+        if (this.type !== supportedTextFieldTypes.NUMBER) return;
+
+        const currentValue = this.value === '' ? 0 : parseFloat(this.value);
+
+        if (isNaN(currentValue)) return;
+
+        const nextStep = this.step + currentValue;
+
+        if (this.max === undefined) {
+            this.value = nextStep;
+            return;
+        }
+
+        if (nextStep > this.max) return;
+
+        if (this.min !== undefined && nextStep < this.min) {
+            this.value = this.min;
+        } else {
+            this.value = nextStep;
+        }
+    }
+
+    /**
+     * Will bump down the input value based on the min and max restrinctions
+     */
+    stepDown() {
+        if (this.type !== supportedTextFieldTypes.NUMBER) return;
+
+        const currentValue = this.value === '' ? 0 : parseFloat(this.value);
+
+        if (isNaN(currentValue)) return;
+
+        const nextStep = currentValue - this.step;
+
+        if (this.min === undefined) {
+            this.value = nextStep;
+            return;
+        }
+
+        if (nextStep < this.min) return;
+
+        if (this.max !== undefined && nextStep > this.max) {
+            this.value = this.max;
+        } else {
+            this.value = nextStep;
+        }
```

Comment:
Same.

Reply:
Fixed

Reply:
Created `setNewStep(nextStep)` and reused. Commenting just for visibility.